test/dhtest.c

```diff
@@ -511,6 +514,8 @@ static int run_rfc5114_tests(void)
 
         Z1 = OPENSSL_malloc(DH_size(dhA));
         Z2 = OPENSSL_malloc(DH_size(dhB));
+        if ((!Z1) || (!Z2))
+            goto bad_err;
```

Comment:
I've added memory free function at bad_err and err process, in this way, memory leak will have no way to exist, thanks:)